skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillsServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.User.UserModel;
+import com.RDS.skilltree.User.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class SkillsServiceImpl implements SkillsService{
+    private final SkillRepository skillRepository;
+    private final UserRepository userRepository;
+    public SkillsServiceImpl(SkillRepository skillRepository,
+                             UserRepository userRepository){
+        this.userRepository = userRepository;
+        this.skillRepository = skillRepository;
+    }
+
+    @Override
+    public SkillDTO getSkillById(UUID id){
+        Optional<SkillModel> skillModel = skillRepository.findById(id);
+        return skillModel.map(SkillDTO::getSkillsWithUsers).orElse(null);
+    }
+
+    @Override
+    public SkillDTO  getSkillByName(String skillName){
+        Optional<SkillModel> skillModel = skillRepository.findByName(skillName);
+        return skillModel.map(SkillDTO::getSkillsWithUsers).orElse(null);
+    }
+
+    @Override
+    public Page<SkillDTO> getAllSkills(Pageable pageable){
+        Page<SkillModel> skillModels = skillRepository.findAll(pageable);
+        return skillModels.map(SkillDTO::getSkillsWithUsers);
+    }
+
+    @Override
+    public SkillDTO createSkill(SkillDRO skillDRO){
+        SkillModel newSkill = SkillDRO.toModel(skillDRO);
+        newSkill.setCreatedAt(Instant.now());
+        newSkill.setUpdatedAt(Instant.now());
+        UserModel user = userRepository.findById(skillDRO.getCreatedBy()).get();
```

Comment:
Is this not Optional? What if the user is not found?

Reply:
User will always exists, as the auth PR ensures this, if the user is found we will store in skill tree db and on next occurances we will find the user.

Reply:
Hi @Ajeyakrishna-k in this PR comment here we are calling the `get()` method right so that is the Optional<> get method only, its just we are extracting the underlying model and storing in the user.

```
        UserModel user = userRepository.findById(skillDRO.getCreatedBy()).get();
        ```